src/Compilers/Test/Core/SourceGeneration/TestSourceGenerator.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Roslyn.Test.Utilities.TestGenerators;
+
+internal class TestSourceGenerator : ISourceGenerator
```

Comment:
We already have this same concept more or less here:

https://github.com/dotnet/roslyn/blob/ec38db3913ef430bcef5097f384c4331fb9db274/src/Compilers/Test/Core/SourceGeneration/TestGenerators.cs#L78

Can we just use that one?

Reply:
Yeah, it's similar but I prefer the pattern of TestSourceGenerator -- the name is easier to find and it is simpler. I can keep TestSourceGenerator in EnC tests, where it's already used before and use CallbackGenerator in compiler tests though.